dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityPreviewServlet.java

```diff
@@ -0,0 +1,359 @@
+import com.dotcms.rendering.velocity.viewtools.content.ContentMap;
+import com.dotcms.visitor.business.VisitorAPI;
+
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.beans.Identifier;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.business.web.HostWebAPI;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.filters.CMSFilter;
+import com.dotmarketing.filters.Constants;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.htmlpageasset.model.IHTMLPage;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.UtilMethods;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.URLDecoder;
+import java.util.Map;
+
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.velocity.Template;
+import org.apache.velocity.context.Context;
+import org.apache.velocity.exception.MethodInvocationException;
+import org.apache.velocity.exception.ParseErrorException;
+import org.apache.velocity.exception.ResourceNotFoundException;
+
+import com.liferay.portal.language.LanguageException;
+import com.liferay.portal.language.LanguageUtil;
+import com.liferay.portal.model.User;
+
+public class VelocityPreviewServlet extends HttpServlet {
+
+    private static final long serialVersionUID = 1L;
+
+
+    private static PermissionAPI permissionAPI = APILocator.getPermissionAPI();
+
+
+
+    private static HostWebAPI hostWebAPI = WebAPILocator.getHostWebAPI();
+
+    private static VisitorAPI visitorAPI = APILocator.getVisitorAPI();
+
+    public static void setPermissionAPI(PermissionAPI permissionAPIRef) {
+        permissionAPI = permissionAPIRef;
+    }
+
+    private String CHARSET = "UTF-8";
+
+
+
+    public static final String VELOCITY_CONTEXT = "velocityContext";
+
+
+    @CloseDBIfOpened
+    protected void service(HttpServletRequest req, HttpServletResponse response) throws ServletException, IOException {
+
+        final String uri = URLDecoder.decode((req.getAttribute(Constants.CMS_FILTER_URI_OVERRIDE) != null)
+                ? (String) req.getAttribute(Constants.CMS_FILTER_URI_OVERRIDE)
+                : req.getRequestURI(), "UTF-8");
+
+
+
+        RequestWrapper request = new RequestWrapper(req);
+        request.setRequestUri(uri);
+
+
+        Host host = hostWebAPI.getCurrentHostNoThrow(request);
+
+        PageMode mode = PageMode.get(request);
+        // Checking if host is active
+        boolean hostlive;
+
+
+        try {
+            hostlive = APILocator.getVersionableAPI()
+                .hasLiveVersion(host);
+        } catch (Exception e1) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)